L1Trigger/L1TMuonOverlap/src/OMTFReconstruction.cc

```diff
@@ -71,7 +72,15 @@ void OMTFReconstruction::beginRun(edm::Run const& run, edm::EventSetup const& iS
 
   m_OMTFConfig->configure(omtfParams);
   m_OMTF->configure(m_OMTFConfig, omtfParams);
-  m_GhostBuster.setNphiBins(m_OMTFConfig->nPhiBins());
+  //m_GhostBuster.setNphiBins(m_OMTFConfig->nPhiBins());
+
+  if(m_Config.exists("ghostBusterType") ) {
+    if(m_Config.getParameter<std::string>("ghostBusterType") == "GhostBusterPreferRefDt")
+    	m_GhostBuster.reset(new GhostBusterPreferRefDt(m_OMTFConfig) );
```

Comment:
@kbunkow How is this memory dealocated

Reply:
@rekovic 
 m_GhostBuster is std::unique_ptr<IGhostBuster> (L1Trigger/L1TMuonOverlap/interface/OMTFReconstruction.h line 77), so the memory is deallocated automatically  when OMTFReconstruction is destructed.

Reply:
@kbunkow Ah, yes.

Reply:
Thx!